src/EVault/shared/BalanceUtils.sol

```diff
@@ -68,6 +68,8 @@ abstract contract BalanceUtils is Base {
     }
 
     function transferBalance(address from, address to, Shares amount) internal virtual {
+        if (to == address(0)) revert E_BadSharesReceiver();
```

Comment:
This check is inconsistent with `validateTransferFromAccount` (using hardcoded `address(0)` instead of constant and allows `address(1)`), is that intentional?